### PR TITLE
map.remove_all implementation

### LINKED ISSUE #520: removeAll implementation [API-1499]
hello can you please implement removeAll for python client [here is the link in java client](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/map/IMap.java#L456) 
thank you very much

### PR DESCRIPTION
**Pull request by feysahin**
map.remove_all method is introduced and implemented with following details:

- remove_all public API and implementation are added
- remove_all code sample is added
- related unittests are added to map_test

Closes #520

### CODE REVIEW

File: examples/map/map_remove_all_example.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import hazelcast
+
+from hazelcast.predicate import between
+
+client = hazelcast.HazelcastClient()
+
+predicate_map = client.get_map("predicate-map").blocking()
+
+for i in range(10):
+    predicate_map.put("key" + str(i), i)
```
Comment (yuce) [inline examples/map/map_remove_all_example.py:10]: How about using f-strings: `predicate_map.put(f"key{i}", i)`

File: hazelcast/proxy/map.py [hunk 2]
```diff
@@ -1283,6 +1284,28 @@ def remove(self, key: KeyType) -> Future[typing.Optional[ValueType]]:
...
+
+        Returns:
+            None
+        """
+
+        check_not_none(predicate, "predicate can't be None")
+
+        try:
+            predicate_data = self._to_data(predicate)
+
```
Comment (yuce) [inline hazelcast/proxy/map.py:1301]: Could you remove the unnecessary blank lines in this method?

**Review by yuce — COMMENTED** [review]
Looks good! Just two minor ideas...

File: hazelcast/proxy/map.py [hunk 3]
```diff
@@ -1283,6 +1284,28 @@ def remove(self, key: KeyType) -> Future[typing.Optional[ValueType]]:
 
         return self._remove_internal(key_data)
 
+    def remove_all(self, predicate: Predicate) -> Future[None]:
```
Comment (mdumandag) [inline hazelcast/proxy/map.py:1287]: From the Javadocs:
```
     * Note that calling this method also removes all entries from
     * caller's Near Cache.
```

So, I believe we should have some code changes in the near cached map as well

File: hazelcast/proxy/map.py [hunk 4]
```diff
@@ -1283,6 +1284,28 @@ def remove(self, key: KeyType) -> Future[typing.Optional[ValueType]]:
...
+        """Removes all entries which match with the supplied predicate
+
+        Args:
+            predicate: Used to select entries to be removed from map.
+
+        Returns:
+            None
```
Comment (mdumandag) [inline hazelcast/proxy/map.py:1294]: I guess we can remove this part, as it is visible from the method signature

File: hazelcast/proxy/map.py [hunk 5]
```diff
@@ -2215,6 +2238,9 @@ def remove(  # type: ignore[override]
     ) -> typing.Optional[ValueType]:
         return self._wrapped.remove(key).result()
 
+    def remove_all(self, predicate: Predicate) -> None:  # type: ignore[override]
+        return self._wrapped.remove_all(predicate).result()
```
Comment (mdumandag) [inline hazelcast/proxy/map.py:2242]: Java does this instead, to clear the near cache

```
    @Override
    protected void removeAllInternal(Predicate predicate) {
        try {
            super.removeAllInternal(predicate);
        } finally {
            nearCache.clear();
        }
    }
```

File: hazelcast/protocol/codec/map_remove_all_codec.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+# hex: 0x013E00
+_REQUEST_MESSAGE_TYPE = 81408
+# hex: 0x013E01
+_RESPONSE_MESSAGE_TYPE = 81409
+
+_REQUEST_INITIAL_FRAME_SIZE = REQUEST_HEADER_SIZE
+
+
+def encode_request(name, predicate):
```
Comment (mdumandag) [inline hazelcast/protocol/codec/map_remove_all_codec.py:13]: Can you also create a PR in the protocol repo with the changes you did to generate this codec?

**Review by yuce — APPROVED** [review]
Thanks @feysahin for your work. We will add the required near cache changes in another PR.

File: hazelcast/protocol/codec/map_remove_all_codec.py (hunk 6 above)
Comment (feysahin) [reply hazelcast/protocol/codec/map_remove_all_codec.py:13]: Sure, I created👍

**Pull request merged**